message_test.go

```diff
@@ -32,20 +32,33 @@ func TestPrivateMessage(t *testing.T) {
 		// ignore response from testClient
 		_, err := bot.PrivateMessage(c.M)
 
-		body, _ := ioutil.ReadAll(bot.Client.(*testClient).Request.Body)
-		if string(body) != c.Body {
-			t.Errorf("got %q, expected %q, case %q", string(body), c.Body, num)
+		// no error expected
+		if err != nil && c.E == nil {
+			t.Fatalf("got %q, expected nil, case %q", err, num)
 		}
 
-		// no error expected
-		if c.E == nil && err != nil {
-			t.Errorf("got %q, expected nil, case %q", err, num)
+		// Check if error matches the error specified in the case
+		switch c.E {
+		case nil:
+			if err != nil && c.E == nil {
+				t.Fatalf("got %q, expected nil, case %q", err, num)
+			}
+
+		default:
+			if err == nil {
+				t.Fatalf("got nil, expected %q, case %q", err, c.E, num)
+			}
+
+			if err.Error() != c.E.Error() {
+				t.Fatalf("got %q, expected %q, case %q", err, c.E, num)
+			}
+
+			return
```

Comment:
minor nitpick: Could you add a comment above the `return` for why the default case returns early?

Maybe something like "return early because no request body will be set in the test client"?